src/alvarium/annotators/handler/ed25519.py

```diff
@@ -0,0 +1,40 @@
+import datetime
+from typing import List
+
+from requests import Request
+from alvarium.annotators.handler.interfaces import RequestHandler
+
+from alvarium.sign.contracts import SignInfo
+from alvarium.sign.ed25519 import Ed25519SignProvider
+from io import StringIO
+from .utils import parseSignature
+
+
+class Ed25519RequestHandler(RequestHandler):
+
+    def __init__(self, request: Request) -> None:
+        self.request = request
+
+    def AddSignatureHeaders(self, ticks: datetime, fields: List[str], keys: SignInfo) -> None:
+        headerValue = StringIO()
+        
+        for i in range(len(fields)):
+            headerValue.write(f'"{str(fields[i])}"')
+            if i < len(fields) - 1:
+                headerValue.write(f' ')
+
+        headerValue.write(f';created={str(int(ticks.timestamp()))};keyid="{str(keys.public.path)}";alg="{str(keys.public.type)}";')
+
+        self.request.headers['Signature-Input'] = headerValue.getvalue()
+        
+        parsed = parseSignature(r=self.request)
+        inputValue = bytes(parsed.seed, 'utf-8')
+        p = Ed25519SignProvider() 
```

Comment:
use the [`SignProviderFactory`](https://github.com/project-alvarium/alvarium-sdk-python/blob/main/src/alvarium/sign/factories.py) to instantiate an  `Ed25519SignProvider` instance. Do not directly invoke the constructor, delegate the instantiation to the factory class. 

Also, you can lift up the instance of the `Ed25519SignProvider` to be a member of the `Ed25519RequestHandler` class, instead of instantiating an object every time you call this method. This can enable you to inject the `SignProvider` from the constructor rather than instantiating an object of `SigneProvider` inside this class. This is an optional change, it's your call whether to implement it or not.